webpack-server.config.js

```diff
@@ -0,0 +1,95 @@
+const EventEmitter = require('events');
+const path = require('path');
+const semver = require('semver');
+const { DefinePlugin } = require('webpack');
+const WebpackAssetsManifest = require('webpack-assets-manifest');
+
+const { AsyncHookPlugin, BuildHookPlugin, transformManifest } = require('./scripts/webpack');
+const { babelLoaderRule, baseConfig, libraryEntries, libraryName, srcPath } = require('./webpack-common.config');
+
+const eventEmitter = new EventEmitter();
+const version = require('./package.json').version;
+const versionDir = `v${semver.major(version)}`;
+const outputPath = path.join(__dirname, 'dist-server', versionDir);
+
+function getServerConfig() {
+    return {
+        ...baseConfig,
+        entry: libraryEntries,
+        output: {
+            filename: '[name]-[contenthash:8].js',
+            library: libraryName,
+            libraryTarget: 'umd',
+            path: outputPath,
+        },
+        module: {
+            rules: [
+                babelLoaderRule,
+                ...baseConfig.module.rules,
+            ],
+        },
+        plugins: [
+            new WebpackAssetsManifest({
+                entrypoints: true,
+                output: path.join(outputPath, 'manifest.json'),
+                publicPath: path.join(process.env.ASSET_HOST || '__ASSET_HOST__', versionDir, '/'),
+                transform: assets => transformManifest(assets, version),
+            }),
+            new BuildHookPlugin({
+                onDone() {
+                    eventEmitter.emit('library:done');
+                },
+            }),
+        ],
+    };
+}
+
+function getServerLoaderConfig() {
+    return {
+        ...baseConfig,
+        entry: {
+            loader: path.join(srcPath, 'loader.ts'),
+        },
+        output: {
+            filename: '[name].js',
+            library: `${libraryName}Loader`,
+            libraryTarget: 'umd',
+            path: outputPath,
+        },
+        module: {
+            rules: [
+                babelLoaderRule,
+                ...baseConfig.module.rules,
+            ],
+        },
+        plugins: [
+            new AsyncHookPlugin({
```

Comment:
Since all the `AsyncHookPlugin` is doing is making these 2 task run sequentially it might be better to just use named configurations and actually run one after the other.

```sh
"build:cdn": "webpack webpack-server.config.js --config-name cdn",
"build:loader": "webpack webpack-server.config.js --config-name loader",
"build:container": "npm run build:cdn && npm run build:loader",
```

or

```sh
"build:cdn": "webpack webpack-server.config.js --config-name cdn && webpack webpack-server.config.js --config-name loader",
```